src/integer/mat_z/mul.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementation of the [`Mul`] trait for [`MatZ`] values.
+
+use super::MatZ;
+use crate::{
+    integer::VecZ,
+    macros::arithmetics::{
+        arithmetic_trait_borrowed_to_owned, arithmetic_trait_cross_types_borrowed_to_owned,
+        arithmetic_trait_cross_types_mixed_borrowed_owned, arithmetic_trait_mixed_borrowed_owned,
+    },
+    traits::{GetNumColumns, GetNumRows},
+};
+use flint_sys::fmpz_mat::fmpz_mat_mul;
+use std::ops::Mul;
+
+impl Mul for &MatZ {
+    type Output = MatZ;
+
+    /// Implements the [`Mul`] trait for two [`MatZ`] values.
+    /// [`Mul`] is implemented for any combination of owned and borrowed [`MatZ`] and any kind of [`VecZ`].
```

Comment:
VecZ? 
```suggestion
    /// [`Mul`] is implemented for any combination of owned and borrowed [`MatZ`].
```

Reply:
This comment is correct. It's about all the combinations of Multiplication of that type that are implemented.